### PR TITLE
feat: Add configuration-driven video source system

### PR DESCRIPTION
**Pull request by apatole**
Add a generic video source management system that allows adding and managing multiple video sources without code modifications, plus several quality-of-life improvements across the framework.

Core Video Source System:
- Add VideoSourceRegistry for centralized video source management
- Implement auto-detection from WebSocket messages with priority-based routing
- Add generic selector lookup and context-aware frame fetching
- Support unlimited video sources (surgical cameras, OR webcams, microscopes, etc.)
- Create dynamic frame queues and mode switching capabilities
- Add source_type field ('uploaded' vs 'livestream') for frontend integration
- Expose /api/video_sources endpoint for frontend to query available sources

Configuration:
- Add video_sources.yaml for declarative source definitions
- Configure two default sources: surgical camera and operating room webcam
- Enable/disable sources, set priorities, and define auto-detection flags
- Support source_type field to differentiate uploaded videos from live streams

Integration & Improvements:
- Update app.py to initialize and register video sources
- Modify web_server.py for dynamic frame queue management
- Implement generic message handling with auto-detection
- Add AnnotationAgent public API (get_annotations, get_latest_annotation)
- Register annotation_agent as resolvable dependency for other agents
- Increase base_agent server wait timeout to 120s for slower vLLM startup
- Improve error message sanitization (avoid logging huge base64 images)
- Fix agent exclusion loading from global.yaml
- Remove debug flags endpoint (not needed for production)

Frontend Updates:
- Add video source configuration loading from backend
- Implement createFrameMessagePayload() with dynamic flags
- Auto-detect source mode based on source_type (uploaded vs livestream)
- Update frame capture to use correct WebSocket flags per source

Testing:
- Add comprehensive test suite covering all functionality
- Test configuration loading, validation, auto-detection, and routing
- Include edge cases and multi-source scenarios

Documentation:
- Add comprehensive section to main README with quick start and examples
- Document key features, configuration reference, and usage patterns
- Include code examples for adding new sources and switching modes
- Document new source_type field in configuration reference
- Update all YAML examples to show source_type usage

Benefits:
- Zero-code extension: Add sources via YAML only
- Auto-discovery: Automatically finds configurations and selectors
- Plugin compatible: Works seamlessly with agent plugin system
- Well-tested: Comprehensive test coverage
- Fully documented: Complete integration guide in main README

### CODE REVIEW

File: configs/video_sources.yaml [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Copyright 2026 VLM-Surgical-Agent-Framework Contributors
```
Comment (mingxin-zheng) [inline configs/video_sources.yaml]: We should replace it with the MONAI copyright plate

Comment (apatole) [reply configs/video_sources.yaml]: Yes will do , thanks
